compiler/coreSyn/CoreLint.hs

```diff
@@ -636,7 +636,8 @@ lintRhs bndr rhs
       = addLoc (LambdaBodyOf var) $
         lintBinder LambdaBind var $ \ var' ->
         do { (body_ty, ue) <- lint_join_lams (n-1) tot enforce expr
-           ; return $ (mkLamType var' body_ty, ue) }
+             -- TODO change to: ue' <- checkLinearity ue var'
```

Comment:
I gather that you didn't fix this one on purpose?

Reply:
Yes, this one currently breaks.